mavros_extras/src/plugins/odom.cpp

```diff
@@ -50,6 +49,43 @@ class OdometryPlugin : public plugin::PluginBase {
 
 		// subscribers
 		odom_sub = odom_nh.subscribe("odom", 10, &OdometryPlugin::odom_cb, this);
+
+		//! Map from param string to local MAV_FRAME
+		const std::unordered_map<std::string, MAV_FRAME> lf_map{
+			{ "vision_enu", MAV_FRAME::VISION_ENU },
+			{ "vision_ned", MAV_FRAME::VISION_NED },
+			{ "mocap_enu", MAV_FRAME::MOCAP_ENU },
+			{ "mocap_ned", MAV_FRAME::MOCAP_NED }
+		};
+
+		//! Map from param string to body MAV_FRAME
+		const std::unordered_map<std::string, MAV_FRAME> bf_map{
+			{ "ned", MAV_FRAME::BODY_NED },
+			{ "frd", MAV_FRAME::BODY_FRD },
+			{ "flu", MAV_FRAME::BODY_FLU }
+		};
+
+		auto local_frame_it = lf_map.find(local_frame);
+		auto body_frame_it = bf_map.find(body_frame_orientation);
```

Comment:
Why not to use `mav_frame_from_str()`?

Reply:
Because, as I explained to you, not all `MAV_FRAME` should be available to be passed, only odometry and body frame related ones. So a mapping is much faster than filtering `mav_frame_from_str()`.

Reply:
Ok, but then why you use different strings (lower case, w/o BODY_ prefix)?